test/integration/lib/update-lock.spec.ts

```diff
@@ -281,4 +284,117 @@ describe('lib/update-lock', () => {
+					{},
+					{ cleanup: [path.join(lockdir, '*', '*', '**.lock')] },
+				).enable();
+				// TODO: Update mocha-pod to work with creating empty directories
+				await mkdirp(path.join(lockdir, '1', 'server'));
+				await mkdirp(path.join(lockdir, '1', 'client'));
+				await mkdirp(path.join(lockdir, '2', 'main'));
+			});
+
+			afterEach(async () => {
+				await testFs.restore();
+				await fs.rm(path.join(lockdir, '1'), { recursive: true });
+				await fs.rm(path.join(lockdir, '2'), { recursive: true });
+			});
+
+			it('releases locks for an appId', async () => {
+				// Lock services for appId 1
+				for (const lockPath of serviceLockPaths[1]) {
+					await lockfile.lock(lockPath);
+				}
+				// Sanity check that locks are taken & tracked by Supervisor
+				expect(lockfile.getLocksTaken()).to.deep.include.members(
+					serviceLockPaths[1],
+				);
+				// Release locks for appId 1
+				await updateLock.releaseLock(1);
+				// Locks should have been released
+				expect(lockfile.getLocksTaken()).to.have.length(0);
+				// Double check that the lockfiles are removed
+				expect(
+					await fs.readdir(path.join(lockdir, '1', 'server')),
+				).to.have.length(0);
+				expect(
+					await fs.readdir(path.join(lockdir, '1', 'client')),
+				).to.have.length(0);
+			});
+
+			it('does not error if there are no locks to release', async () => {
+				expect(lockfile.getLocksTaken()).to.have.length(0);
+				// Should not error
+				await updateLock.releaseLock(1);
+				expect(lockfile.getLocksTaken()).to.have.length(0);
+			});
+
+			it('ignores locks outside of appId scope', async () => {
+				const lockPath = path.join(lockdir, '2', 'main', 'updates.lock');
+				// Lock services outside of appId scope
+				await lockfile.lock(lockPath);
+				// Sanity check that locks are taken & tracked by Supervisor
+				expect(lockfile.getLocksTaken()).to.deep.include.members([lockPath]);
+				// Release locks for appId 1
+				await updateLock.releaseLock(1);
+				// Locks for appId 2 should not have been released
+				expect(lockfile.getLocksTaken()).to.deep.include.members([lockPath]);
+				// Double check that the lockfile is still there
+				expect(
+					await fs.readdir(path.join(lockdir, '2', 'main')),
+				).to.have.length(1);
+				// Clean up the lockfile
+				await lockfile.unlock(lockPath);
+			});
+
+			it('waits to release locks until resource write lock is taken', async () => {
+				// Lock services for appId 1
+				for (const lockPath of serviceLockPaths[1]) {
+					await lockfile.lock(lockPath);
+				}
+				// Sanity check that locks are taken & tracked by Supervisor
+				expect(lockfile.getLocksTaken()).to.deep.include.members(
+					serviceLockPaths[1],
+				);
+				// Take the write lock for appId 1
+				const release = await writeLock(1);
+				// Queue releaseLock, won't resolve until the write lock is released
+				const releaseLockPromise = updateLock.releaseLock(1);
+				// Locks should have not been released
```

Comment:
I think you need some delay here. Even if the `releaseLock` call wasn't being blocked, the expectation below would still return as that line would be executed before the promise could be executed.

Reply:
Added 500ms delay [here](https://github.com/balena-os/balena-supervisor/pull/2234/commits/7b9be98a7831ff63764d7a20261881535a0b1d30#diff-e7a2b32a616114c913967bbeb93301d1194c25d37b5c6d6393146a862d33b61fR446) for releaseLock and [here](https://github.com/balena-os/balena-supervisor/pull/2234/commits/fe9f940ce619cd7045cb87bba1f66b2e8f91a93d#diff-e7a2b32a616114c913967bbeb93301d1194c25d37b5c6d6393146a862d33b61fR510) for takeLock